### PR TITLE
add google-one-tap types

### PR DESCRIPTION
**Pull request by voidpumpkin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/google-one-tap/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// Type definitions for google-one-tap 1.0
+// Project: https://developers.google.com/identity/one-tap/web
+// Definitions by: voidpumpkin <https://github.com/voidpumpkin>
+//                 kostasmanionis <https://github.com/kostasmanionis>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace google;
```
Comment (amcasey) [inline types/google-one-tap/index.d.ts:7]: If these types are consumable globally, there should be global tests as well.

https://github.com/definitelyTyped/DefinitelyTyped/#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module

**Review by amcasey — CHANGES_REQUESTED** [review]

**Comment by amcasey** [comment]
Sorry, I actually meant that there should be two sets of tests.

**Comment by voidpumpkin** [comment]
@amcasey Im confused abit, this is not a npm module and the only way you can use this is via global.
Or you mean I should keep that old file ?

**Comment by amcasey** [comment]
@voidpumpkin My mistake, I've been looking at dual-use packages all day.

**Review by amcasey — APPROVED** [review]

**Comment by voidpumpkin** [comment]
Ready to merge

**Pull request merged**